exp/api/v1beta1/azuremachinepool_types.go

```diff
@@ -306,6 +306,11 @@ type (
 		// next reconciliation loop.
 		// +optional
 		LongRunningOperationStates infrav1.Futures `json:"longRunningOperationStates,omitempty"`
+
+		// CustomDataHash saves the hash for the stored custom data to be able to check if we need to update the custom data
+		// in reconciliation loops.
+		// +optional
+		CustomDataHash *string `json:"customDataHash"`
```

Comment:
Would an annotation be a better fit for this?

Reply:
this might be superseded by https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3134#discussion_r1107459190

Reply:
Resolved by using the SDKtoVMSS vs the API